bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -774,16 +774,15 @@ public void collapseAll() {
 	 */
 	public void collapseToLevel(Object elementOrTreePath, int level) {
 		Assert.isNotNull(elementOrTreePath);
-		Control control = getControl();
-		try {
-			control.setRedraw(false);
-			Widget w = internalGetWidgetToSelect(elementOrTreePath);
-			if (w != null) {
-				internalCollapseToLevel(w, level);
+		executeWithRedrawDisabled(new Runnable() {
+			@Override
+			public void run() {
+				Widget w = internalGetWidgetToSelect(elementOrTreePath);
+				if (w != null) {
+					internalCollapseToLevel(w, level);
+				}
```

Comment:
You can simplify this by putting into a lambda expression:
```suggestion
		executeWithRedrawDisabled(() -> {
			Widget w = internalGetWidgetToSelect(elementOrTreePath);
			if (w != null) {
				internalCollapseToLevel(w, level);
			}
```
This also applies to the other instantiations of `Runnable`.